### DISCUSSION TITLE
Profiling actions and loaders

### DISCUSSION THREAD

**Discussion by WesleyYue** [remix-run/remix#5371] [Q&A] [answered]
What is the best way to profile a slow loader function? Ideally, I'm looking for something that either traces code execution, or at least profiles network calls (similar to the network panel in browser dev tools).

**Comment by sergiodxa** [comment, marked as answer]
If you're using Node you can use `node --inspect` and use the Chrome DevTools to profile it

**Reply by WesleyYue** [reply]
Thanks for the tip. For future readers. I ended up using a combination of console.profile() and viewing the profile results in VSCode, as well as an external http proxy inspector (I use Proxyman specifically, but there are a bunch of options)
